### ISSUE TITLE
--cb_type dr gives wrong "average loss"

### ISSUE THREAD

**Issue by marco-rossi29** [VowpalWabbit/vowpal_wabbit#1696]
data.json (see below) contains two events. In both events, action 3 is played (i.e., **"_labelIndex":3**).
DR predicts `action 0` for both events (so the correct average loss = 0) but the output average loss is non-zero (-0.133899), which is wrong.

```
vw --dsjson data.json --cb_adf --cb_type dr -P 1
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = data.json
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
0.000000 0.000000            1            1.0    known        0:0...        8
-0.133899 -0.267798            2            2.0    known        0:-0.265735...        8

finished run
number of examples = 2
weighted example sum = 2.000000
weighted label sum = 0.000000
average loss = -0.133899
total feature number = 16
```

To Repro:
All recent VW versions exibit this issue. Commits tested:
- 8.6.1 commit f085178d12c778024a0e533b5bf7a44a79f3d816
- 8.5.0 commit aff7eb4bcd772b86c0658e42a9ec414c71654ab1
- 8.4.0 commit e2e2cd4cb76f914ced19d98395a0094d7a1787f0

`data.json`: The following two events:
```
{"_label_cost":-1,"_label_probability":0.85,"_label_Action":4,"_labelIndex":3,"o":[{"EventId":"457092bc4357477b92915130cbde04b5-2itjL","v":1}],"Timestamp":"2018-10-14T00:01:17.8660000Z","Version":"1","EventId":"457092bc4357477b92915130cbde04b5-2itjL","a":[4,1,2,3],"c":{"_synthetic":false,"User":{"_age":0},"_multi":[{"action":{"name":1}},{"action":{"name":2}},{"action":{"name":3}},{"action":{"name":4}}]},"p":[0.85,0.05,0.05,0.05],"VWState":{"m":"bf08eb2d7af24abe8248e38021ba0c31-2isRa/6b0a050cb0ad44a1998ab90212a2c91d-2it0G"}}
{"_label_cost":-1,"_label_probability":0.85,"_label_Action":4,"_labelIndex":3,"o":[{"EventId":"32a65eef6f4745149ee5dae7f6c92a3c-2itku","v":1}],"Timestamp":"2018-10-14T00:02:02.9570000Z","Version":"1","EventId":"32a65eef6f4745149ee5dae7f6c92a3c-2itku","a":[4,1,2,3],"c":{"_synthetic":false,"User":{"_age":0},"_multi":[{"action2":{"name2":1}},{"action2":{"name2":2}},{"action2":{"name2":3}},{"action2":{"name2":4}}]},"p":[0.85,0.05,0.05,0.05],"VWState":{"m":"bf08eb2d7af24abe8248e38021ba0c31-2isRa/6b0a050cb0ad44a1998ab90212a2c91d-2it0G"}}
```

**Comment by marco-rossi29** [comment]
The issue that I see here is that average loss produced by cb_type dr is not fully determined by the predictions, and I think we need to fix this.

I see two solutions:
- Ensure that the average loss is the IPS estimate produced using the predictions of a policy (or a metric that depends only on the predictions).
- If we determine that the average loss doesn't only depends on the predictions (and so it is cb_type dependent), then I suggest to add as an additional output of VW the IPS estimate (well defined given the predictions) and then we will use that for offline experimentation (e.g., to compare between cb_types).

**Comment by JohnLangford** [comment]
In theory, DR estimates can be more precise/lower variance.  

There is nothing wrong with outputting both however.

**Comment by marco-rossi29** [comment]
I feel that comparing the "average loss" produced by DR with the "average loss" produced by IPS and/or MTR (even if both estimators are unbiased) is comparing apples with oranges.

I'm fine using a different estimator than IPS for offline experimentation, as long as we use the same estimator across the board. And then we can build confidence intervals around such estimator.

**Comment by JohnLangford** [comment]
I think the issue is that two things are changing at once: the estimator
and the learning algorithm.   In a more ideal approach, these would be
orthogonal so you can mix and match things.  The tricky element here is
that we don't (at present) have a way to do DR estimation without DR
learning.   It's not hard to disable DR estimation with DR learning,
potentially as a flag.

-John

**Comment by jackgerrits** [comment]
This will be addressed by #2222
